packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,349 +7,375 @@ excerpt: >-
-
-@Controller
-public class HomeController {
-
-    @RequestMapping("/")
-    public String home(@AuthenticationPrincipal Saml2AuthenticatedPrincipal principal, Model model) {
-        model.addAttribute("name", principal.getName());
-        model.addAttribute("emailAddress", principal.getFirstAttribute("email"));
-        model.addAttribute("userAttributes", principal.getAttributes());
-        return "home";
-    }
-
-}
-```
-
-Create a `src/main/resources/templates/home.html` file to render the user's information.
-
-```html
-<!DOCTYPE html>
-<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="https://www.thymeleaf.org"
-      xmlns:sec="https://www.thymeleaf.org/thymeleaf-extras-springsecurity6">
-<head>
-    <title>Spring Boot and SAML</title>
-    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
-</head>
-<body>
-
-<h1>Welcome</h1>
-<p>You are successfully logged in as <span sec:authentication="name"></span></p>
-<p>Your email address is <span th:text="${emailAddress}"></span>.</p>
-<p>Your authorities are <span sec:authentication="authorities"></span>.</p>
-<h2>All Your Attributes</h2>
-<dl th:each="userAttribute : ${userAttributes}">
-    <dt th:text="${userAttribute.key}"></dt>
-    <dd th:text="${userAttribute.value}"></dd>
-</dl>
-
-<form th:action="@{/logout}" method="post">
-    <button id="logout" type="submit">Logout</button>
-</form>
-
-</body>
-</html>
-```
-
-Create a `src/main/resources/application.yml` file to contain your metadata URI.
-
-```yaml
-spring:
-  security:
-    saml2:
-      relyingparty:
-        registration:
-          okta:
-            assertingparty:
-              metadata-uri: <your-metadata-uri>
-```
-
-Then, change `build.gradle` to use `thymeleaf-extras-springsecurity6` instead of `thymeleaf-extras-springsecurity5` and add Spring Security SAML's dependency:
-
-```groovy
-implementation 'org.thymeleaf.extras:thymeleaf-extras-springsecurity6'
-implementation 'org.springframework.security:spring-security-saml2-service-provider'
-```
+1. Add `src/main/java/com/example/demo/HomeController.java` to populate the authenticated user's information.
+
+   ```java
+   package com.example.demo;
+
+   import org.springframework.security.core.annotation.AuthenticationPrincipal;
+   import org.springframework.security.saml2.provider.service.authentication.Saml2AuthenticatedPrincipal;
+   import org.springframework.stereotype.Controller;
+   import org.springframework.ui.Model;
+   import org.springframework.web.bind.annotation.RequestMapping;
+
+   @Controller
+   public class HomeController {
+
+      @RequestMapping("/")
+      public String home(@AuthenticationPrincipal Saml2AuthenticatedPrincipal principal, Model model) {
+         model.addAttribute("name", principal.getName());
+         model.addAttribute("emailAddress", principal.getFirstAttribute("email"));
+         model.addAttribute("userAttributes", principal.getAttributes());
+         return "home";
+      }
+
+   }
+   ```
```

Comment:
Java code is typically indented 4 spaces, not 2. Can you please update?

Reply:
Updated in next commit